pkg/gofr/crud_handlers.go

```diff
@@ -148,30 +190,84 @@ func (a *App) registerCRUDHandlers(e *entity, object interface{}) {
 }
 
 func (e *entity) Create(c *Context) (interface{}, error) {
+	newEntity, err := e.bindAndValidateEntity(c)
+	if err != nil {
+		return nil, err
+	}
+
+	fieldNames, fieldValues := e.extractFields(newEntity)
+
+	stmt, err := sql.InsertQuery(c.SQL.Dialect(), e.tableName, fieldNames, fieldValues, e.constraints)
+	if err != nil {
+		return nil, err
+	}
+
+	result, err := c.SQL.ExecContext(c, stmt, fieldValues...)
+	if err != nil {
+		return nil, err
+	}
+
+	var lastID interface{}
+
+	if hasAutoIncrementID(e.constraints) { // Check for auto-increment ID
+		lastID, err = result.LastInsertId()
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		lastID = fieldValues[0]
+	}
+
+	return fmt.Sprintf("%s successfully created with id: %v", e.name, lastID), nil
```

Comment:
I would have returned only lastID

```suggestion
	return lastID, nil
```

The sprintf can be done in the code calling the method as the entity is the receiver here, e.name is available when calling the method

Reply:
returning the **lastID** here is not a very good option as this is the response what user will get by default until the user doesn't override the CRUD methods. The Create is called by POST here:

```go
if fn, ok := object.(Create); ok {
		a.POST(basePath, fn.Create)
	} else {
		a.POST(basePath, e.Create)
	}
```

thatswhy placed at last of Create so if user chooses to follow the default implementation path then it should give a proper response

```go
func (e *entity) Create(c *Context) (interface{}, error) {
	newEntity, err := e.bindAndValidateEntity(c)
	if err != nil {
		return nil, err
	}

	fieldNames, fieldValues := e.extractFields(newEntity)

	stmt, err := sql.InsertQuery(c.SQL.Dialect(), e.tableName, fieldNames, fieldValues, e.constraints)
	if err != nil {
		return nil, err
	}

	result, err := c.SQL.ExecContext(c, stmt, fieldValues...)
	if err != nil {
		return nil, err
	}

	var lastID interface{}

	if hasAutoIncrementID(e.constraints) { // Check for auto-increment ID
		lastID, err = result.LastInsertId()
		if err != nil {
			return nil, err
		}
	} else {
		lastID = fieldValues[0]
	}

	return fmt.Sprintf("%s successfully created with id: %v", e.name, lastID), nil
}
```